lisa/sut_orchestrator/azure/common.py

```diff
@@ -193,25 +206,141 @@ class AzureVmPurchasePlanSchema:
+            )
+        ),
+    )
+    disk_controller_type: Optional[
+        Union[
+            search_space.SetSpace[schema.DiskControllerType], schema.DiskControllerType
+        ]
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[schema.DiskControllerType.SCSI, schema.DiskControllerType.NVME],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_nullable_set_space,
+                base_type=schema.DiskControllerType,
+                is_allow_set=True,
+                default_values=[
+                    schema.DiskControllerType.SCSI,
+                    schema.DiskControllerType.NVME,
+                ],
+            )
+        ),
+    )
+    hyperv_generation: Optional[
+        Union[search_space.SetSpace[int], int]
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[1, 2],
+        ),
+        metadata=field_metadata(
+            decoder=partial(search_space.decode_set_space_by_type, base_type=int)
+        ),
+    )
     network_data_path: Optional[
         Union[search_space.SetSpace[schema.NetworkDataPath], schema.NetworkDataPath]
     ] = field(  # type: ignore
         default_factory=partial(
             search_space.SetSpace,
+            is_allow_set=True,
             items=[
                 schema.NetworkDataPath.Synthetic,
                 schema.NetworkDataPath.Sriov,
             ],
         ),
         metadata=field_metadata(
             decoder=partial(
-                search_space.decode_set_space_by_type, base_type=schema.NetworkDataPath
+                search_space.decode_set_space_by_type,
+                base_type=schema.NetworkDataPath,
             )
         ),
     )
+    security_profile: Union[
+        search_space.SetSpace[SecurityProfileType], SecurityProfileType
+    ] = field(  # type:ignore
+        default_factory=partial(
+            search_space.SetSpace,
+            is_allow_set=True,
+            items=[
+                SecurityProfileType.Standard,
+                SecurityProfileType.SecureBoot,
+                SecurityProfileType.CVM,
+                SecurityProfileType.Stateless,
+            ],
+        ),
+        metadata=field_metadata(
+            decoder=partial(
+                search_space.decode_nullable_set_space,
+                base_type=SecurityProfileType,
+                is_allow_set=True,
+                default_values=[
+                    SecurityProfileType.Standard,
+                    SecurityProfileType.SecureBoot,
+                    SecurityProfileType.CVM,
+                    SecurityProfileType.Stateless,
+                ],
+            )
+        ),
+    )
+
+    def load_features(self, platform: "AzurePlatform") -> None:
```

Comment:
`load_from_platform`. The "feature" is used as a key concept in LISA. Avoid using it in other ways.